src/core/contracts/subscription.ts

```diff
@@ -0,0 +1,124 @@
+import { HexString } from "../../types/Strings";
+import { ethers } from "ethers";
+import { requireGetProvider } from "../../config";
+import { dsnpBatchFilter } from "./announcement";
+import { Filter } from "@ethersproject/abstract-provider";
+import { Announcer__factory } from "../../types/typechain";
+const ANNOUNCER_DECODER = new ethers.utils.Interface(Announcer__factory.abi);
+
+/**
+ * BatchAnnounceCallbackArgs: interface for callback function that is passed to batchAnnounceEvents
+ */
+export interface BatchAnnounceCallbackArgs {
+  blockNumber: number;
+  transactionHash: HexString;
+  dsnpType: number;
+  dsnpUri: string;
+  dsnpHash: HexString;
+}
+
+interface ParsedLog {
+  fragment: ethers.utils.LogDescription;
+  log: ethers.providers.Log;
+}
+
+interface batchFilterOptions {
+  dsnpType?: number;
+  startBlock?: number;
+}
+type BatchAnnounceCallback = (doReceiveAnnouncement: BatchAnnounceCallbackArgs) => void;
+
+/**
+ * batchAnnounceEvents() sets up a listener to listen to retrieve Batch Announce events from the chain.
+ * It takes a callback and a filter. The filter is used to filter events that come through.
+ * The callback is invoked for each correctly filtered event.
+ *
+ * @param callback - The callback function to be called when an event is received
+ * @param filters -  Any filter options for including or excluding certain events
+ * @returns        A function that can be called to remove listener for this type of event
+ */
+export const batchAnnounceEvents = async (
```

Comment:
Can we rename this to something like `subscribeToBatchAnnounceEvents` or just `subscribe`? This naming makes it seem like this method is for announcing a batch.

Reply:
Also, depending on whether "batch" is a verb or a noun here, this may be breaking our function naming convention.

Reply:
It is in the subscription module. But I do see your point. I can do that.